src/workerd/io/actor-cache.h

```diff
@@ -599,6 +638,18 @@ class ActorCache final: public ActorCacheInterface {
     auto begin() { return inner.begin(); }
     auto end() { return inner.end(); }
 
+    // The way we coalesce reads means we update `Entry`s while they're still in the `dirtyList`.
+    // This means the size of the list may change, without us adding or removing an Entry.
+    // To ensure proper accounting, we have to modify the size when we modify an Entry.
```

Comment:
Heh, this seems like the kind of logic that will lead to a bug that breaks proper accounting. Again, my fault for not yet having read the implementation (so this is largely a note for me to come back to), but why are we updating the size of the entry *while* it's in the dirty list rather than when it's being removed from the dirty list (because the read is done and thus it no longer needs to be in the dirty list)?

Reply:
Would you want it to happen as part of the actual removal from the list (i.e. `removeEntry()` or `DirtyList::remove()`?

I think I would find it less difficult to track a down a bug if we explicitly modified the size _prior_ to attempting to remove from `dirtyList` because we would know it's limited to get() operations rather than any arbitrary removal from `dirtyList`.

Reply:
That's what I would have expected, but you have a point, and at this point I don't remember the context enough to have a strong opinion. It's fine to leave as is.